build dependency rename not respected

Repro:

https://github.com/shikhar/rules-rust-playground/tree/main/value-bag

`bazel run main`

```
Starting local Bazel server and connecting to it...
INFO: Analyzed target //:main (39 packages loaded, 1776 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_sbhushan/5bde5da6fd341b8f812bc9cb64ebc180/external/crates__value_bag__1_0_0_alpha_7/BUILD.bazel:29:19: Compiling Rust bin value_bag_build_script_ v1.0.0-alpha.7 (16 files) failed: (Exit 1): process_wrapper failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/rules_rust/util/process_wrapper/process_wrapper --subst 'pwd=${pwd}' -- external/rust_darwin_x86_64/bin/rustc ... (remaining 22 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
error[E0433]: failed to resolve: use of undeclared crate or module `rustc`
 --> external/crates__value_bag__1_0_0_alpha_7/build.rs:4:8
  |
4 |     if rustc::is_feature_flaggable().unwrap_or(false) {
  |        ^^^^^ use of undeclared crate or module `rustc`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
Target //:main failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /private/var/tmp/_bazel_sbhushan/5bde5da6fd341b8f812bc9cb64ebc180/external/crates__value_bag__1_0_0_alpha_7/BUILD.bazel:66:13 Compiling Rust rlib value_bag v1.0.0-alpha.7 (16 files) failed: (Exit 1): process_wrapper failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/rules_rust/util/process_wrapper/process_wrapper --subst 'pwd=${pwd}' -- external/rust_darwin_x86_64/bin/rustc ... (remaining 22 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
INFO: Elapsed time: 5.637s, Critical Path: 0.16s
INFO: 4 processes: 4 internal.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

The `value-bag` crate has `version_check` as a build dependency, renamed to `rustc`:
https://github.com/sval-rs/value-bag/blob/864bba3c3b62480a6049c99e6ebd0fb25d4d1bae/Cargo.toml#L100-L102

It also has other renamed regular dependencies, which work fine on older versions of the crate where this build dependency did not exist (https://github.com/shikhar/rules-rust-playground/commit/a7c85ed734943c246515ad88552d1635aa391dd7)

--- a/crate_universe/src/templates/partials/build_script.template
+++ b/crate_universe/src/templates/partials/build_script.template
@@ -51,6 +51,7 @@ cargo_build_script(
     {%- include "templates/partials/targeted_build_script_data_dependencies.template" -%},
     {%- else -%},
     {%- endif %}
+    aliases = {%- include "templates/partials/targeted_aliases.template" %},
     tags = [
         "cargo-raze",
         "manual",

--- a/examples/crate_universe/basic/BUILD.bazel
+++ b/examples/crate_universe/basic/BUILD.bazel
@@ -7,6 +7,7 @@ rust_binary(
     edition = "2018",
     deps = [
         crate("lazy_static"),
+        crate("value-bag"),
     ],
 )
 

--- a/examples/crate_universe/basic/src/main.rs
+++ b/examples/crate_universe/basic/src/main.rs
@@ -1,5 +1,6 @@
 use lazy_static::lazy_static;
 use std::collections::HashMap;
+use value_bag::ValueBag;
 
 lazy_static! {
     static ref HASHMAP: HashMap<&'static str, &'static str> = {
@@ -14,4 +15,9 @@ lazy_static! {
 fn main() {
     assert_eq!(HASHMAP["Daniel"], "Wagner-Hall");
     println!("It worked!");
+
+
+    let bag = ValueBag::capture_display(&42);
+    let num = bag.to_u64().unwrap();
+    println!("hello {}", num);
 }

--- a/examples/crate_universe/basic/src/main.rs
+++ b/examples/crate_universe/basic/src/main.rs
@@ -13,10 +13,10 @@ lazy_static! {
 }
 
 fn main() {
-    assert_eq!(HASHMAP["Daniel"], "Wagner-Hall");
-    println!("It worked!");
-
     let bag = ValueBag::capture_display(&42);
     let num = bag.to_u64().unwrap();
-    println!("hello {}", num);
+
+    assert_eq!(num, 42);
+    assert_eq!(HASHMAP["Daniel"], "Wagner-Hall");
+    println!("It worked!");
 }

--- a/examples/crate_universe/basic/src/main.rs
+++ b/examples/crate_universe/basic/src/main.rs
@@ -16,7 +16,6 @@ fn main() {
     assert_eq!(HASHMAP["Daniel"], "Wagner-Hall");
     println!("It worked!");
 
-
     let bag = ValueBag::capture_display(&42);
     let num = bag.to_u64().unwrap();
     println!("hello {}", num);

--- a/examples/crate_universe/basic/workspace.bzl
+++ b/examples/crate_universe/basic/workspace.bzl
@@ -10,6 +10,10 @@ def deps():
                 name = "lazy_static",
                 semver = "=1.4",
             ),
+            crate.spec(
+                name = "value-bag",
+                semver = "=1.0.0-alpha.7",
+            ),
         ],
         supported_targets = [
             "x86_64-apple-darwin",

